### ISSUE TITLE
Kover CLI report generation does not reliably with --excludeAnnotation when using JAR (instead of folder) for --classfiles

### ISSUE THREAD

**Issue by oliviernotteghem** [Kotlin/kotlinx-kover#709]
We found instances where, when using --classfiles to point to a JAR, --excludeAnnotation does not work completely and some code that should be excluded still appear to be covered.

With the same command line have --classfiles pointing to a folder, when the JAR was unpacked, the filtering works reliably.

Example of command line invocation :
```java -jar /Users/oliviern/.devxp/import-cache/4b81c0960d39ddd828f97e31aee273bf0d929d87/3c34899b72c05e31efaa6ed042ad0527573034a1/kover-cli-0.8.2.jar @bazel-bin/libraries/foundation/ui/ui-compose-ambients/test_release-kover_metadata.txt --xml bazel-bin/libraries/foundation/ui/ui-compose-ambients/test_release-kover_report.xml --html /Users/oliviern/Uber/android/bazel-testlogs/libraries/foundation/ui/ui-compose-ambients/test_release/coverage_report```

content of bazel-bin/libraries/foundation/ui/ui-compose-ambients/test_release-kover_metadata.txt:

```
report
bazel-out/xplat-fastbuild/bin/libraries/foundation/ui/ui-compose-ambients/test_release-kover_report.ic
--title
Code-Coverage Analysis: @@//libraries/foundation/ui/ui-compose-ambients:test_release
--src
libraries/foundation/ui/ui-compose-ambients/src/main/kotlin/com/uber/ui/compose/core/ambient
--classfiles
bazel-out/xplat-fastbuild/bin/libraries/foundation/ui/ui-compose-ambients/libsrc_release.jar
--exclude
*Expose

```
example of code failing to be filtered out:

```
  @Expose
  fun composeDeps(
      analyticsClient: AnalyticsClient,
  ) =
      object : composeDependencies {
        override fun analyticsClient() = analyticsClient
      }
```

**Comment by zuevmaxim** [comment]
@oliviernotteghem Could you please look into the bytecode of the class inside jar, and check that the annotation is apesent there?
Unpack the jar, and then use `javap -v -l -p <path-to-class>.class` to see the bytecode. The target method should have the annotation. 
While testing I found out that ProGuard removed `RetentionPolicy.CLASS` annotation in my case, maybe it is something similar in your case

**Comment by oliviernotteghem** [comment]
@zuevmaxim : I forgot to mention that we verified this already. The annotation is not stripped and is present in bytecode.

**Comment by shanshin** [comment]
@oliviernotteghem, to reproduce it locally we need reproducer project or a problematic jar file

**Comment by oliviernotteghem** [comment]
Sure. Here is problematic jar 
[src_main.jar.zip](https://github.com/user-attachments/files/17984883/src_main.jar.zip) and .ic file 
[test_main-kover_report.ic.zip](https://github.com/user-attachments/files/17984896/test_main-kover_report.ic.zip)

The code corresponds to the screenshot below. 
![Screenshot 2024-12-02 at 2 59 18 PM](https://github.com/user-attachments/assets/628def20-c97e-4f87-957f-5d1e862f1003)

During report generation, we use `--excludeAnnotation *Expose`, however you can it worked on expose() method, but not for the code in composeDeps()

**Comment by shanshin** [comment]
Fixed in `0.9.0`

**Comment by oliviernotteghem** [comment]
Thank you very much. We upgraded to 0.9.0 and can confirm the issue is gone. There seems to be other fixes as well as our code coverage for failing module is not non-null, which materializes into a nice code coverage increase for Uber android monorepo :)
